### ISSUE TITLE
Add bitdepth functionality

### ISSUE THREAD

**Issue by gekoke** [nwg-piotr/nwg-displays#35]
I'd like to specify the bitdepth for outputs using this program. Both sway and Hyprland have support for this in their configs.

**Comment by nwg-piotr** [comment]
I'll take a look at this soon.

**Comment by nergdron** [comment]
I'd love this as well, my main display supports 10bit output, and right now I have to manually edit `monitors.conf` to re-enable it after using `nwg-displays`.

**Comment by nwg-piotr** [comment]
We'll think about it next year. ;)

**Comment by nwg-piotr** [comment]
It seems we'll need some own data file, which is what I wanted to avoid. Or maybe you guys know how to check if 10-bit depth is turned on per display/monitor? Neither `swaymsg -t get_outputs` nor `hyprctl monitors` seems to return this info.

**Comment by nergdron** [comment]
since I use `hyprland`, I've filed an upstream bug for them to provide this, since I think that's probably the best solution. if someone who uses sway could do something similar, that'd be rad:

https://github.com/hyprwm/Hyprland/issues/4971

**Comment by nwg-piotr** [comment]
Thank you.

**Comment by nwg-piotr** [comment]
As it has been already done in Hyprland, I'll add a form field soon. I only need to get us rid of wlr-randr, as vaxerski made it possible in the same commit. Either way, both changes won't work on Hyprland <= 0.36.0.

**Comment by nergdron** [comment]
@nwg-piotr will the `hyprland-git` package in arch get me on that dev branch, or will I have to build from source manually?

**Comment by nwg-piotr** [comment]
hyprland-git is what you need. :)

**Comment by nergdron** [comment]
perfect. i'll try to get to that soon, i'm in the middle of week 3 of covid, so i'm not exactly running at full processing power.

**Comment by nwg-piotr** [comment]
No hurry. We won't release this before hyprland > 0.36.0, while Arch is still o 0.35.0. Take care and get well soon.

**Comment by gekoke** [comment]
Awesome - thanks for your work!

**Comment by nergdron** [comment]
this new build doesn't run for me, crashes with the following. are we waiting on some dev release tools that aren't available in AUR yet?

```
Settings: {'view-scale': 0.25, 'snap-threshold': 10, 'indicator-timeout': 500, 'custom-mode': [], 'use-desc': False}
Running on Hyprland
Traceback (most recent call last):
  File "/usr/bin/nwg-displays", line 33, in <module>
    sys.exit(load_entry_point('nwg-displays==0.3.15', 'gui_scripts', 'nwg-displays')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/nwg_displays/main.py", line 1252, in main
    create_display_buttons()
  File "/usr/lib/python3.11/site-packages/nwg_displays/main.py", line 569, in create_display_buttons
    outputs = list_outputs()
              ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/nwg_displays/tools.py", line 146, in list_outputs
    for item in m["availableModes"]:
                ~^^^^^^^^^^^^^^^^^^
KeyError: 'availableModes'
```

**Comment by nwg-piotr** [comment]
@nergdron Are you on Hyprland 0.37.1? The `hyprctl monitors` command does not return available modes in Hyprland <= 0.36.

**Comment by nwg-piotr** [comment]
The disabled one is what crashes the program. I'll fix it tonight.

**Comment by nwg-piotr** [comment]
The crash on the 'monitor' key has been reported in #54. Let's continue up there.
